src/extra/widgets/chart/lv_chart.c

```diff
@@ -571,7 +567,7 @@ void lv_chart_set_value_by_id(lv_obj_t * obj, lv_chart_series_t * ser, uint16_t
 
     if(id >= chart->point_cnt) return;
     ser->y_points[id] = value;
-    lv_chart_refresh(obj);
+    invalidate_point(obj, id);
```

Comment:
I think for the `LV_CHART_TYPE_BAR` we need to add this line:
`invalidate_point(obj, id + 1);`
too but I'm not sure, I would appreciate you @kisvegabor to check it.

Reply:
I've tested and it worked for me `invalidate_point(obj, id);`. Why would it be required to use `+1`?

Reply:
> I've tested and it worked for me with `invalidate_point(obj, id);`. Why would it be required to use `+1`?

I'm saying it, just because of the way we do it in `lv_chart_set_next_value`, as you know we invalidate 2 times. Not sure why!

Reply:
Hm, `lv_example_chart_2()` seems working well if `lv_chart_set_next_value` is modified like:
```c
    lv_chart_t * chart  = (lv_chart_t *)obj;
    ser->y_points[ser->start_point] = value;
    invalidate_point(obj, ser->start_point);
    ser->start_point = (ser->start_point + 1) % chart->point_cnt;
//    invalidate_point(obj, ser->start_point);
//    lv_chart_refresh(obj);
```

And it seems logical too. Can you confirm it?

Reply:
I tested `lv_example_chart_2()`, but even when I comment the first `invalidate_point` it still works!
```c
lv_chart_t * chart  = (lv_chart_t *)obj;
    ser->y_points[ser->start_point] = value;
 //   invalidate_point(obj, ser->start_point);
    ser->start_point = (ser->start_point + 1) % chart->point_cnt;
//    invalidate_point(obj, ser->start_point);
```
can't you remember why the second `invalidate_point` was added before?

Reply:
It's not working for me with no `invalidate_point`. Let use one `invalidate_point` and merge it to see if any issues comes up.

Reply:
O.K, I removed it. So if it is O.K please merge.🙏